jsonnet/kube-prometheus/kube-prometheus-strip-limits.libsonnet

```diff
@@ -0,0 +1,56 @@
+// Strips spec.containers[].limits for certain containers
+// https://github.com/coreos/kube-prometheus/issues/72
+{
```

Comment:
contained within this mixin I think we can have a config object for each container's resources

Reply:
Sure! Something like this?

```jsonnet
  _config+:: {
    nodeExporter+: {
      containerLimits+: {
        kubeRbacProxy+: {
          cpu: 0,
          memory: 0,
        },
        nodeExporter+: {
          cpu: 0,
          memory: 0,
        },
      },
    },
    prometheusOperator+: {
      containerLimits: {
        configReloader: {
          cpu: 0,
          memory: 0,
        },
      },
    },
    kubeStateMetrics+: {
      containerLimits: {
        addonResizer: {
          cpu: 0,
          memory: 0,
        },
        kubeRbacProxyMain: {
          cpu: 0,
          memory: 0,
        },
        kubeRbacProxySelf: {
          cpu: 0,
          memory: 0,
        },
      },
    },
  },
```

Or perhaps all the RBAC proxies can be configured in one place?

Reply:
> Or perhaps all the RBAC proxies can be configured in one place?

Yes I think that would be appropriate.

I was more imagining something along the lines of this for the configuration:

```jsonnet
  _config+:: {
    resources+:: {
      "node-exporter": {}, // container name as key, so we can just set the resources to what matches here, but by "default" set all of them to empty so by default the patch removes all resource requests and limits
      "kube-rbac-proxy": {
        requests: {cpu: "100m", "50Mi"}, // example demonstration of how we would set requests
      },
    },
  },
```

What do you think?